cleverhans/attacks.py

```diff
@@ -13,18 +15,26 @@ class Attack(object):
 
     def __init__(self, model, back='tf', sess=None):
         """
-        :param model: A function that takes a symbolic input and returns the
-                      symbolic output for the model's predictions.
+        :param model: An instance of the cleverhans.Model class, or optionally
+                      (but not recommended) a function that takes a symbolic
+                      input and returns the symbolic output for the model's
+                      predictions.
         :param back: The backend to use. Either 'tf' (default) or 'th'.
         :param sess: The tf session to run graphs in (use None for Theano)
         """
         if not(back == 'tf' or back == 'th'):
             raise ValueError("Backend argument must either be 'tf' or 'th'.")
         if back == 'th' and sess is not None:
             raise Exception("A session should not be provided when using th.")
-        if not hasattr(model, '__call__'):
-            raise ValueError("model argument must be a function that returns "
-                             "the symbolic output when given an input tensor.")
+        if not isinstance(model, Model):
+            if hasattr(model, '__call__'):
+                warnings.warn("The model argument should be an instance of the"
```

Comment:
What would you think about just declaring this interface deprecated and issuing a warning that it will be gone in 6 months?